src/app/rpmostree-builtin-kargs.c

```diff
@@ -160,42 +253,90 @@ rpmostree_ex_builtin_kargs (int            argc,
     return EXIT_FAILURE;
 
   /* We extract the existing kernel arguments from the boot configuration */
-  const char *existing_kernel_arg_string = NULL;
+  const char *old_kernel_arg_string = NULL;
   if (!g_variant_lookup (boot_config, "options",
-                         "&s", &existing_kernel_arg_string))
+                         "&s", &old_kernel_arg_string))
     return EXIT_FAILURE;
 
   if (display_kernel_args)
     {
       g_print("The kernel arguments are:\n%s\n",
-              existing_kernel_arg_string);
+              old_kernel_arg_string);
       return EXIT_SUCCESS;
     }
 
   g_autofree char *transaction_address = NULL;
   char *empty_strv[] = {NULL};
 
-  /* dbus does not allow NULL to mean the empty string array,
-   * assign them to be empty string array here to prevent erroring out
-   */
-  if (!opt_kernel_replace_strings)
-    opt_kernel_replace_strings = empty_strv;
-  if (!opt_kernel_append_strings)
-    opt_kernel_append_strings = empty_strv;
-  if (!opt_kernel_delete_strings)
-    opt_kernel_delete_strings = empty_strv;
-
-  /* call the generearted dbus-function */
-  if (!rpmostree_os_call_kernel_args_sync (os_proxy,
-                                           existing_kernel_arg_string,
-                                           (const char* const*) opt_kernel_append_strings,
-                                           (const char* const*) opt_kernel_replace_strings,
-                                           (const char* const*) opt_kernel_delete_strings,
-                                           get_kargs_option_variant (),
-                                           &transaction_address,
+  if (opt_editor)
+    {
+      /* We track the kernel arg instance before the editor */
+      const char *sysroot_path = rpmostree_sysroot_get_path (sysroot_proxy);
+      g_autoptr(GFile) sysroot_file = g_file_new_for_path (sysroot_path);
+      g_autoptr(OstreeSysroot) before_sysroot = ostree_sysroot_new (sysroot_file);
+      if (!ostree_sysroot_load (before_sysroot, cancellable, error))
+        return FALSE;
+
+      const char* current_kernel_arg_string = NULL;
+      if (!kernel_arg_handle_editor (old_kernel_arg_string, &current_kernel_arg_string,
+                                     cancellable, error))
+        return EXIT_FAILURE;
+      gboolean out_changed = FALSE;
+
+      /* Here we load the sysroot again, if the sysroot
+       * has been changed since then, we directly error
+       * out.
+       */
+      if (!ostree_sysroot_load_if_changed (before_sysroot,
+                                           &out_changed,
                                            cancellable,
                                            error))
-    return EXIT_FAILURE;
+        return EXIT_FAILURE;
+      if (out_changed)
+        {
+          g_set_error (error, G_IO_ERROR, G_IO_ERROR_INVALID_ARGUMENT,
+                       "Bootconfig has already been changed since then, please retry the editor");
```

Comment:
How about "Conflict: bootloader configuration changed"?

Reply:
Also IMO we should avoid deleting the temp file in this case.  See for example what I [do in git-evtag](https://github.com/cgwalters/git-evtag/blob/d32fdef6694d1a47965c6d5c1c50b8d2bf87f95e/src/git-evtag.c#L725).  (I *had* to add this as sometimes my Yubikey Neo isn't initialized, and I'd write a long git tag message only to have it be discarded when `gpg2` failed to sign it)

Reply:
sure, now in the fixup, we tend to rewrite user's input into a new temp file created. The reason for doing that is so we do not need to change the original code copied from `ot-editor.c`. 

A cleaner way(I think) would be to directly modify the editor function copied from `ot-editor.c` to also return the file (and not deleting it).  but yea... change shared code. seems not that neat (I feel). 

So, If there is other easier implementations, please let me know =)